src/lib/core/a11y/focus-trap.ts

```diff
@@ -47,6 +49,12 @@ export class FocusTrap {
       return root;
     }
 
+    let focusTarget = root.querySelector(FOCUS_TARGET_SELECTOR) as HTMLElement;
```

Comment:
We don't want to call `querySelector` here- this method is used recursively, you would would potentially end up calling `querySelector` many times. This should happen outside of the recursive method.